src/JSONPrinting.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+#include "KafkaMessageMetadataStruct.h"
+#include <algorithm>
+#include <nlohmann/json.hpp>
+#include <yaml-cpp/yaml.h>
+
+class JSONPrinting {
```

Comment:
I don't think there is any reason for this to be a class. You can use free functions instead and put the private methods in an anonymous namespace instead. Then `yaml` include can also be moved to the `.cpp` file.